## 3.1 
📂 Daten einlesen und vorbereiten

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

In [2]:
games = pd.read_csv("/Users/ylli/Library/CloudStorage/OneDrive-AlteKantonsschuleAarau/BBBaden/M259/games.csv")
games.head()

AppID                   Name Release date Estimated owners  Required age  \
0    20200       Galactic Bowling    21-Oct-08        0 - 20000             0   
1   655370           Train Bandit    12-Oct-17        0 - 20000             0   
2  1732930           Jolt Project    17-Nov-21        0 - 20000             0   
3  1355720               Henosis™    23-Jul-20        0 - 20000             0   
4  1139950  Two Weeks in Painland     3-Feb-20        0 - 20000             0   

   Price                                Supported languages  Positive  \
0  19.99                                        ['English']         6   
1   0.99  ['English', 'French', 'Italian', 'German', 'Sp...        53   
2   4.99                 ['English', 'Portuguese - Brazil']         0   
3   5.99  ['English', 'French', 'Italian', 'German', 'Sp...         3   
4   0.00                     ['English', 'Spanish - Spain']        50   

   Negative  Achievements  Recommendations  Average playtime forever  \
0        11            30                0                         0   
1         5            12                0                         0   
2         0             0                0                         0   
3         0             0                0                         0   
4         8            17                0                         0   

              Publishers                                         Categories  \
0  Perpetual FX Creative  Single-player,Multi-player,Steam Achievements,...   
1           Wild Rooster  Single-player,Steam Achievements,Full controll...   
2          Campião Games                                      Single-player   
3      Odd Critter Games              Single-player,Full controller support   
4          Unusual Games                   Single-player,Steam Achievements   

                            Genres  
0              Casual,Indie,Sports  
1                     Action,Indie  
2  Action,Adventure,Indie,Strategy  
3           Adventure,Casual,Indie  
4                  Adventure,Indie

In [3]:
X = games.select_dtypes(include=["number"])
y = games["Average playtime forever"]  # Zielvariable
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

## 3.2 
🎯 Algorithmus wählen und Modell trainieren

Ich wähle die Lineare Regression, da "Average playtime forever" eine kontinuierliche, numerische Zielvariable ist.    
Sie ermöglicht es, einen klaren, linearen Zusammenhang zwischen Eingabe- und Zielvariablen zu modellieren und liefert dabei nachvollziehbare, interpretierbare Ergebnisse.  
Gerade bei überschaubaren und sauberen Datensätzen wie dem vorliegenden eignet sich die Lineare Regression hervorragend, um erste belastbare Vorhersagen zu erstellen und das Verhalten der Daten besser zu verstehen.

In [4]:
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

## 3.3 
🧪 Vorhersagen generieren und Modell testen

In [5]:
y_pred = model.predict(X_test)

results = pd.DataFrame({
    'Spielnummer': [1, 2, 3, 4, 5],
    'Tatsächliche Spielzeit (min)': [y_test.iloc[i] for i in range(5)],
    'Vorhergesagte Spielzeit (min)': [round(y_pred[i], 2) for i in range(5)]
})

results

Spielnummer  Tatsächliche Spielzeit (min)  Vorhergesagte Spielzeit (min)
0            1                             0                            0.0
1            2                             0                           -0.0
2            3                           249                          249.0
3            4                             0                           -0.0
4            5                             0                           -0.0

Ich habe die ersten 5 Vorhersagen manuell mit den tatsächlichen Werten verglichen.  
Alle Vorhersagen stimmen exakt oder nahezu exakt mit den tatsächlichen Werten überein.  
Insbesondere zeigt sich bei Spielen mit einer durchschnittlichen Spielzeit von 0 Minuten, dass das Modell korrekt 0 vorhergesagt hat.  
Auch der höhere Wert von 249 Minuten wurde genau getroffen.  
Insgesamt funktioniert das Modell sehr zuverlässig und liefert sinnvolle, präzise Vorhersagen für die durchschnittliche Spielzeit.

## 3.4 
📝 Zusammenfassung schreiben

Zusammenfassung:
Das Modell der linearen Regression liefert eine solide Vorhersage für die durchschnittliche Spielzeit.
Der R2-Score zeigt, wie gut die Varianz der tatsächlichen Werte erklärt wird. 
Einzelne Abweichungen sind sichtbar, jedoch ist die Tendenz der Vorhersagen sinnvoll und realistisch.
Traditionell bleibt die Lineare Regression eine bewährte Methode für einfachere Regressionsaufgaben.
